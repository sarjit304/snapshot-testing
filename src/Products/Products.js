function Products({items = []}) {
    if (items.length <= 0) {
        return <div>No items found.</div>
    } else {
        return <ol>
            {items.map((item, index) => {
                return <li key={index}>{item}</li>
            })}
        </ol>
    }
}

export default Products;