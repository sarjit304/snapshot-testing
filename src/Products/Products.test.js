import renderer from 'react-test-renderer';
import Products from './Products';

test('renders when there are no items', () => {
    const tree = renderer.create(<Products/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('renders when there is one item', () => {
    const tree = renderer.create(<Products items={["Earth"]}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('renders when there are multiple item', () => {
    const tree = renderer.create(<Products items={['Earth', 'Sun', 'Moon', 'Saturn']}/>).toJSON();
    expect(tree).toMatchSnapshot();
})