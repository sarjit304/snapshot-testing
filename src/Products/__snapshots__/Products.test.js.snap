// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are multiple item 1`] = `
<ol>
  <li>
    Earth
  </li>
  <li>
    Sun
  </li>
  <li>
    Moon
  </li>
  <li>
    Saturn
  </li>
</ol>
`;

exports[`renders when there are no items 1`] = `
<div>
  No items found.
</div>
`;

exports[`renders when there is one item 1`] = `
<ol>
  <li>
    Earth
  </li>
</ol>
`;
